### ISSUE TITLE
Terminology & UI used in Mapping work flow

### ISSUE THREAD

**Issue by philbarker** [t3-innovation-network/desm#624]
**General**
* Some places we have Elements, others we have Property: use "Property/Element" throughout. 

* Make sure that the "DSO" name is not being used anywhere, just the "Specification/Schema Name".

**First step:** "Upload Specification"  /new-mapping

![Image](https://github.com/user-attachments/assets/5e347eb9-862b-4412-9dad-544b17cf14e8)

* "Name of your specification" should be marked mandatory

* "Which **domain** are you uploading?" should be "Which **abstract class** are you mapping?"

* "**Domains** in green have a spine..." should be "**Abstract classes** in green have a spine...."

* Guidance on file types should read:
"You can upload your specification as RDF (Turtle, JSON-LD or RDF/XML),  JSON Schema, XML Schema (XSD), or CSV."

**Pop-up screen, when file is uploaded:**

![Image](https://github.com/user-attachments/assets/22100219-a888-4948-9998-bd90006fceb5)

* Should be "classes" for "domains" throughout.

**Preview of imported properties**

![Image](https://github.com/user-attachments/assets/fab4569f-3b16-4a19-8662-88b0a075c999)

* "Add Vocabulary" should be "Add Controlled Vocabulary"

 ... and the dialog that follows 

![Image](https://github.com/user-attachments/assets/daebc6b5-543b-440a-b98f-e4e293de7385)

will change as we implement more flexibility for that, but: 
* terminology "Controlled Vocabulary" not just "Vocabulary" should be maintained; 
* the supported formats are: RDF - SKOS (Turtle, JSON-LD or RDF/XML serializations), JSON-Schema and XML Schema (XSD) enumerations, and CSV.

**Second step** Domain mapping /mappings/302/map
For this step we are mapping Classes in the uploaded schema to Abstract Classes in the Abstract Domain Model; but the classes are "Abstract Classes" not Domains, and we are only working on one of them, so:

![Image](https://github.com/user-attachments/assets/c9fa465f-630f-4e3e-a5d6-b4ae5802b328)

* "Map to Domains" (in the progress bar) should be  "Map Classes" (because "Map Classes to Abstract Domain Model" is too long, but if we have mouse-over hints that should be it). 

* The buttons that say "Done Domain Mapping" can just say "Done".

* Please swap the "Save" button and "Done" button, so that Done is on the right (this puts save on the same side in all steps.

* "2. Add the properties to the proper domain" should be "2. Add the relevant properties/elements to this abstract class".  This should be the heading (as was "Upload Specification" for step 1).

* The current heading "Mapping DESM Demo and Documentation - Organization - test" (which varies) should be broken down explicitly, in this case as:
**Mapping for:** DESM Demo and Documentation.  [that's the mapping project/config profile name]
**Abstract Class:** Organization.   [the abstract class we're working on]
**Schema:** test.  [the name of the schema]

* "0 of 1 property added to domains" should be "0 of 1 property added to abstract class"

* "Drag a matching property here" should be "Drag relevant properties/elements here".

* "Now you can drag and drop them to the matching domain on the left " (in the red box on the right) should be "Now you can select the relevant properties/elements for this abstract class."

**Third step** Mapping to spine mappings/302/align

![Image](https://github.com/user-attachments/assets/c86d3745-3c0f-40f4-bfb9-9220a3af6a15)

* "Align and fine tune" isn't what anyone calls this. It should be  "Map Properties" (because "Map Classes to Spine for Abstract Class" is too long, but if we have mouse-over hints that should be it).

* Some places we have Elements, others we have Property: use "Property/Element" throughout 

Note: "Domains" is fine in a property definition.

* I think on this page the "DSO" name is being used not the specification name.  On the previous page I was mapping "Test", on this page I see "CredReg" and "CTDL". It should be the specification name.

* "Add Synthetic" should be "Add Spine Term" 

When I click to add a new term to the spine: 

![Image](https://github.com/user-attachments/assets/4b5c290d-ecc5-4f63-ad94-b2aa1d76a59f)

* "Synthetic property added to spine" should be "New term added to spine".

**the view of all mappings /mappings**

![Image](https://github.com/user-attachments/assets/883fc758-c65e-4dc6-9ad1-d8fbddf8c072)

* Let's call this page "Mappings for «Project Name»"
* Let's default to "all mappings" with the filter for "my mappings" as the choice that may be selected.
* Break what is currently shown as the "Specification name" into Abstract Class and Specification, add the Version to the Specification (to be clear it is not the version of the mapping). Prepend the Specification Name with the relevant DSO.  [Note that right now we do not know what the DSO name is as the organization name we record in the mapping project config profiles is used as the affiliation of the mappers, so this will be blank for now.] 
*  The status should mirror the wording used in the steps/progress indicator at the top of the page, that is one of: Ready to Import; Mapping Classes; Mapping Properties; Published.

The page and column headings would then be something like:

## Mappings for DESM Demo and Documentation ##

| Abstract Class | DSO: Specification (Version) | Mapped |	Status	| Author (Affiliation) |
|---------------------|---------------------|---------------------|---------------------|---------------------|
| Person | Spine
| Organization | Spine					
| Person name | Spine
| Organization | DSO: Spec name (ver 1)| 1/1 | Mapping Properties | Phil Barker (Cetis) |

**Comment by philbarker** [comment]
This suggestion from a previous user, to allow access to controls at top of element/property mapping page when working down the list of properties/elements to be mapped:

![Image](https://github.com/user-attachments/assets/739b7083-fed5-4273-ac0d-25f25c13187a)

**Comment by philbarker** [comment]
We should use some term more general than "Domain" and "Range" when describing the class a property can be used with and what type of value is expected. Not many people know what they are, and they only apply to RDF. The following comes from a JSON-Schema, which doen't have domains and ranges:
![Image](https://github.com/user-attachments/assets/b2a4defd-ce3b-48f7-9664-2f59afd83c68)

I suggest
* Used on class(es): 
* Expected value type(s):

[Though the first of those supposes "classes" which XML doesn't really have, so I'll take suggestions if anyone has one.]

**Comment by philbarker** [comment]
On the home page (from #600):

We need to refresh the home page in a way that avoids showing redundant and duplicate buttons and to improve clarity of what each button does.

The current home page for someone who is not logged in show:
 
![image](https://github.com/user-attachments/assets/dd459a0c-0d16-4472-a5ec-0d98259cbe04)

There is a lot of variation in how the buttons are shown: why are some on the top and others on the side? why are some white-on-black others white?  I don't think this helps. 

The text under "Map your Schema" is (I think) the only place we use the word schema; we use specification pretty much everywhere else. Besides, nothing here is relevant if you are not logged in (and most viewers of this page will not be mappers, we hope)

* The only buttons that actually work when not logged in are "View Shared Mappings" and "Sign In".  Please show only this button as active at this stage. Other buttons should not be shown until the user is logged in. 

* The "View Specifications" block on the left should have the heading be "View Published Mappings". The should read "View Mappings" There is no need for further explanation.  Is it possible just to list the mappings here, rather than have a button that links to a drop down list? Either way the  list should be asking to "Choose Mapping" not "Choose Configuration Profile".

* "Map your schema" should be "Map your specification" The text under it should say "If you have an account you may log in to map your specification."

* There should be no "New Mapping" button here.  Let's keep the buttons for logged-in users at the top.


When I log in as a standard user, I see

![image](https://github.com/user-attachments/assets/bc44db42-3abd-4b66-8cef-79ddf88dd983)

* The "View Mappings" at the top needs to be renamed, as  "My Mappings", to distinguish it from "View Shared / Published Mappings".  It should be a black button just like the others.

* The "New Mapping" button at the top should be black.

I think we only need one New Mapping button. 

* Remove the section "Map your schema" from the left of the body of the page. The information currently under "Map Your Schema" should be on the page that you get to after hitting the "New Mapping" button, not here.

* The View Specification should have the heading be "View Published Mappings", see above, and should not change from how it is when the user is not logged in.

When I sign in as an Admin User, the only change is in the right of the top bar, where I get a button to link to the dashboard 
![image](https://github.com/user-attachments/assets/8b2501f3-5ec6-4b9f-a092-ce01e81acf3a)

I think it would be better not to spread the button that link to different activities all over the interface like this.  I always have to look for this button, and access to it is the only reason why I would log in as admin. 

* Please put the "Dashboard" button with the other buttons (My Mappings, New Mapping).

**Comment by philbarker** [comment]
[I know this contradicts what I wrote above last month, sorry seeing it in real life gave me second thoughts.]

When I sign in as a mapper I see:

<img width="1326" height="827" alt="Image" src="https://github.com/user-attachments/assets/3013294e-a7e7-4134-a60f-d65fdb710392" />

Note that it shows me as signed in but still has the text saying that I may log in to map my specification in the lower left.  Would it be possible to change this to have links to the My Mappings and New Mappings pages (same as the links at the top, which need to be there on all screens when logged in including this one); so at the bottom left, for logged in users only we would have:

**Map Your Specification**
* [My mappings](https://gopher.desm.staging.c66.me/mappings)
* [New mapping](https://gopher.desm.staging.c66.me/new-mapping)

(keep the text for users who are not logged in as it is)

**Comment by shlag3n** [comment]
sure, will update later today or on Monday.

**Comment by shlag3n** [comment]
Do we need to update home page for logged in admin user too?

**Comment by philbarker** [comment]
Thanks @shlag3n !  Yes, I think is should be updated for logged-in admins too (I think they can do mapping now, right?)

**Comment by philbarker** [comment]
OK, admin users cannot map, so the text when logged in as an admin user user should say:

**Map Your Specification**
You are currently logged in as an admin user and so cannot map specifications. 


Also remove the "My Mappings" and "New Mapping" button from the top for admin users, as they won't work.
